Support for Azure Cache for Redis resource type (#595)
tomkerkhove/promitor: docs/configuration/metrics/redis-cache.md

---
layout: default
title: Azure Cache for Redis Declaration
---

> ## Azure Cache for Redis - ![Availability Badge](https://img.shields.io/badge/Available%20Starting-v0.1.0-green.svg)

Reviewer: It's actually only available starting v1.0

```suggestion
## Azure Cache for Redis - ![Availability Badge](https://img.shields.io/badge/Available%20Starting-v1.0.0-green.svg)
```

Author: done.

You can declare to scrape an Azure Cache for Redis via the `RedisCache` resource type.

The following fields need to be provided:

> - `cacheName` - The name of the Redis Cache service

Reviewer: ```suggestion
- `cacheName` - The name of the Redis Cache instance
```

Author: done.

> The list of supported metrics is available on the Azure Cache for Redis [monitoring documentation page](https://docs.microsoft.com/en-us/azure/azure-cache-for-redis/cache-how-to-monitor#available-metrics-and-reporting-intervals).

Reviewer: Maybe good to point to official overview for Monitor to get all metrics, with a pointer to Redis docs for more info?
```suggestion
All supported metrics are documented in the official [Azure Monitor documentation](https://docs.microsoft.com/en-us/azure/azure-monitor/platform/metrics-supported#microsoftcacheredis).

You can find more documentation on each metric in the Azure Cache for Redis [monitoring documentation](https://docs.microsoft.com/en-us/azure/azure-cache-for-redis/cache-how-to-monitor#available-metrics-and-reporting-intervals).
```

Author: yes, totally makes sense.

Example:
```yaml
name: redis_cache_hits
description: "The number of successful key lookups during the specified reporting interval. This maps to keyspace_hits from the Redis INFO command."
resourceType: RedisCache

> cacheName: cachejco

Reviewer: Let's change this to Promitor
```suggestion
cacheName: cachejco
```

Author: done.

scraping:
  schedule: "0 */2 * ? * *"
azureMetricConfiguration:
  metricName: CacheHits
  aggregation:

>     type: Average

Reviewer: Let's align with https://docs.microsoft.com/en-us/azure/azure-monitor/platform/metrics-supported#microsoftcacheredis
```suggestion
    type: Total
```

Author: done.
